blocks/components/editable/tinymce.js

```diff
@@ -19,12 +22,42 @@ export default class TinyMCE extends wp.element.Component {
 		}
 	}
 
-	shouldComponentUpdate() {
+	componentDidMount() {
+		this.initialize();
+	}
+
+	shouldComponentUpdate( nextProps ) {
 		// We must prevent rerenders because TinyMCE will modify the DOM, thus
 		// breaking React's ability to reconcile changes.
 		//
 		// See: https://github.com/facebook/react/issues/6802
-		return false;
+		return nextProps.tagName !== this.props.tagName;
+	}
+
+	componentWillUpdate( nextProps ) {
+		if ( ! this.editor ) {
+			return;
+		}
+
+		if ( nextProps.tagName !== this.props.tagName ) {
+			this.editor.destroy();
+			this.editor = null;
+		}
+	}
+
+	componentDidUpdate( prevProps ) {
+		if ( this.props.tagName !== prevProps.tagName ) {
+			this.initialize();
+		}
+	}
+
+	componentWillUnmount() {
+		if ( ! this.editor ) {
```

Comment:
Dunno if it's worth the abstraction, but I could see value in a `destroy` method which handles (a) verifying editor instance property, (b) destroying editor instance, (c) unsetting instance property:

```js
destroy() {
	if ( this.editor ) {
		this.editor.destroy();
		delete this.editor;
	}
}
```

To the point of `delete` vs `= null;`, I don't feel strongly one way or the other, but I think in cases where we assign any instance property, it would be a good idea to initialize it to its default value in `constructor`.